modules/v2/main.tf

```diff
@@ -94,18 +120,21 @@ resource "google_cloud_run_v2_service" "main" {
     }
 
     dynamic "containers" {
-      for_each = var.containers
+      for_each = concat(var.containers,
+      var.enable_prometheus_sidecar ? local.prometheus_sidecar_container : [])
```

Comment:
IMO we should create a separate containers block for creating prometheus sidecar?  WDYT @bharathkkb!

Reply:
@q2w any reason for that? This essentially adding to dynamic list of blocks so not sure if there is any benefit to separate block?

Reply:
Since the sidecar containers block requires very configuration to be provided. It seemed cleaner to me to keep a simple separate containers block. I don't see any other benefit. This approach also works! Resolving this comment!

```
containers {
    container_name  = "collector"
    container_image = "us-docker.pkg.dev/cloud-ops-agents-artifacts/cloud-run-gmp-sidecar/cloud-run-gmp-sidecar:1.1.1"
    depends_on = [ list of containers ]
}
```